### PR TITLE
DAOS-6533 control: Automatically update fabric provider

### PR DESCRIPTION
**Pull request by kjacque**
Previously updating the fabric provider required reformatting the entire system. With this change, the fabric provider can be changed by stopping all engines, shutting down all daos_servers, and changing the provider in the config files. The MS leader is considered the canonical source of the system's expected fabric provider. It will reject ranks that attempt to join with unexpected fabric providers.

Features: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by mjmac — APPROVED** [review]

**Comment by kjacque** [comment]
I just noticed an issue in the `engine_join_failed` RAS event. It is not including the correct address for the machine. I will re-push. I've also written up a section in the troubleshooting guide discussing how to detect and resolve the case where the engine failed to join due to the fabric provider.

**Review by knard-intel — APPROVED** [review]
Mostly OK for me (just minor formatting documentation detail).
From my understanding, this feature will not be used so often but will be very helpful for support tasks when needed.
Thus, it could be relevant to have at least one dedicated functional test checking this new feature.

File: docs/admin/common_tasks.md [hunk 1]
```diff
@@ -108,3 +108,13 @@ differ by at least 100.
...
+
+## Change fabric provider on a DAOS system
+
+1. Stop all DAOS client I/O.
+1. Evict client pool handles to ensure I/O has stopped.
+1. Shut down all `daos_server` processes.
+1. Update the fabric provider and interfaces in all `daos_server` configuration files. All `daos_server` configurations must use the same fabric provider.
+1. Restart all `daos_server` processes to re-load the configuration file.
+1. Ensure all ranks have re-joined by running `dmg system query`. If some ranks fail to join, check logs to troubleshoot.
+1. After all ranks have joined, restart `daos_agent` processes on client nodes, or alternately send SIGUSR2 to all of these processes to refresh network information.
```
[the remark this answers is not in the document]
Comment (mjmac) [reply docs/admin/common_tasks.md:120]: Using `1.` in a numbered list is valid Github Markdown.

You can verify that it renders correctly by viewing the rich diff, e.g. https://github.com/daos-stack/daos/pull/13784/files?short_path=faecaed#diff-faecaeda0051255c3748894cb7e66502cc158a356c1f6f3bbbca830f16331a2a

[the remark this answers is not in the document]
Comment (knard-intel) [reply docs/admin/common_tasks.md:120]: Thanks for the tips, not aware of that :)

Comment (mjmac) [inline docs/admin/common_tasks.md:120]: Hmm. Just occurred to me, but we probably also need to tell the admin to restart their agents in order to pick up the new URIs, right?

Comment (kjacque) [reply docs/admin/common_tasks.md:120]: That's what I'm suggesting on that line of the doc. Unless you think it needs to be added elsewhere?

**Comment by kjacque** [comment]
> Thus, it could be relevant to have at least one dedicated functional test checking this new feature.

@jamesanunez is working on adding the ftests for this feature.

File: docs/admin/common_tasks.md (hunk 1 above)
Comment (mjmac) [reply docs/admin/common_tasks.md:120]: ... I don't know how I misread that! 😵 Sorry for the noise, LGTM.

**Review by mjmac — APPROVED** [review]

**Pull request merged**